File: 13-02-2026/problem_2_array_rotation/correct/array_rotation.c
// PROBLEM 2: ARRAY ROTATION - CORRECT C
// =============================================
#include <stdio.h>

int main() {
    int n;
    long long k;
    scanf("%d %lld", &n, &k);
    int a[200000];
    
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    
    k = k % n;
    
    for (int i = 0; i < n; i++) {
        int idx = (i - k + n) % n;
        printf("%d", a[idx]);
        
        if (i == n - 1)
            printf("\n");
        else
            printf(" ");
    }
    
    return 0;
}


// ===================================================

File: 13-02-2026/problem_2_array_rotation/correct/array_rotation.py
# PROBLEM 2: ARRAY ROTATION - CORRECT PYTHON
# =============================================
import sys

n, k = map(int, sys.stdin.readline().split())
a = list(map(int, sys.stdin.readline().split()))

k %= n

for i in range(n):
    idx = (i - k + n) % n
    if i == n - 1:
        print(a[idx])
    else:
        print(a[idx], end=" ")


# ===================================================